lib/portlayer/storage/nfs/volume.go

```diff
@@ -272,3 +274,37 @@ func (v *VolumeStore) getMetadata(op trace.Operation, ID string, target Target)
 	op.Infof("Successfully read volume metadata at (%s)", metadataPath)
 	return info, nil
 }
+
+func (v *VolumeStore) StatPath(op trace.Operation, deviceId string, targetPath string) (*compute.FileStat, error) {
+	target, err := v.Service.Mount(op)
+	if err != nil {
+		return nil, err
+	}
+	defer v.Service.Unmount(op)
+
+	files, err := target.ReadDir(v.volDirPath(deviceId))
+	var result string
+	for _, file := range files {
+		result = result + " " + file.Name()
+	}
+
+	fileInfo, _, err := target.Lookup(path.Join(v.volDirPath(deviceId), targetPath))
+	if err != nil {
+		return nil, err
+	}
+
+	// nfs server returns "" for file name when doing a look up, need to fill in the value from targetPath instead
+	// trimsuffix shouldn't be needed since the statpath personality should trim the suffix
+	newPath := targetPath
+	for strings.HasSuffix(newPath, "/") {
```

Comment:
this could also be `for trimmed := strings.TrimSuffix(newpath, "/"); trimmed != newpath {newpath = trimmed}` 

I am not sure what exactly is being achieved here( I may be missing something jlmk). but I think you could also just do a `_, filename := filepath.Split(targetPath)` if all you want is the name of the file here.